drivers/frequency/adf4377/adf4377.c

```diff
@@ -0,0 +1,674 @@
+{
+	uint8_t buff[ADF4377_BUFF_SIZE_BYTES];
+
+	if (dev->spi_desc->bit_order) {
+		buff[0] = bit_swap_constant_8(reg_addr);
+		buff[1] = bit_swap_constant_8(ADF4377_SPI_WRITE_CMD);
+		buff[2] = bit_swap_constant_8(data);
+	} else {
+		buff[0] = ADF4377_SPI_WRITE_CMD;
+		buff[1] = reg_addr;
+		buff[2] = data;
+	}
+
+	return spi_write_and_read(dev->spi_desc, buff, ADF4377_BUFF_SIZE_BYTES);
+}
+
+/**
+ * @brief Update ADF4377 register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Mask for specific register bits to be updated.
+ * @param data - Data read from the device.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+int32_t adf4377_spi_write_mask(struct adf4377_dev *dev, uint8_t reg_addr,
+			       uint8_t mask, uint8_t data)
+{
+	uint8_t read_val;
+	int32_t ret;
+
+	ret = adf4377_spi_read(dev, reg_addr, &read_val);
+	if (ret < 0)
+		return ret;
+
+	read_val &= ~mask;
+	read_val |= data;
+
+	ret = adf4377_spi_write(dev, reg_addr, read_val);
+
+	return ret;
+}
+
+/**
+ * @brief Reads data from ADF4377 over SPI.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param data - Data read from the device.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+int32_t adf4377_spi_read(struct adf4377_dev *dev, uint8_t reg_addr,
+			 uint8_t *data)
+{
+	int32_t ret;
+	uint8_t buff[ADF4377_BUFF_SIZE_BYTES];
+
+	if (dev->spi_desc->bit_order) {
+		buff[0] = bit_swap_constant_8(reg_addr);
+		buff[1] = bit_swap_constant_8(ADF4377_SPI_READ_CMD);
+		buff[2] = bit_swap_constant_8(ADF4377_SPI_DUMMY_DATA);
+	} else {
+		buff[0] = ADF4377_SPI_READ_CMD;
+		buff[1] = reg_addr;
+		buff[2] = ADF4377_SPI_DUMMY_DATA;
+	}
+
+	ret = spi_write_and_read(dev->spi_desc, buff, ADF4377_BUFF_SIZE_BYTES);
+	if(ret < 0)
+		return ret;
+
+	*data = buff[2];
+
+	return ret;
+}
+
+/**
+ * @brief ADF4377 SPI Scratchpad check.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t adf4377_check_scratchpad(struct adf4377_dev *dev)
+{
+	int32_t ret;
+	uint8_t scratchpad;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x0A), ADF4377_SPI_SCRATCHPAD_A);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_read(dev, ADF4377_REG(0x0A), &scratchpad);
+	if (ret < 0)
+		return ret;
+
+	if(scratchpad != ADF4377_SPI_SCRATCHPAD_A)
+		return FAILURE;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x0A), ADF4377_SPI_SCRATCHPAD_B);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_read(dev, ADF4377_REG(0x0A), &scratchpad);
+	if (ret < 0)
+		return ret;
+
+	if(scratchpad != ADF4377_SPI_SCRATCHPAD_B)
+		return FAILURE;
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set default datasheet register values.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+static int32_t adf4377_set_default(struct adf4377_dev *dev)
```

Comment:
Is this a default configuration or does it write the initial register value to the registers?